src/transformers/models/univnet/modeling_univnet.py

```diff
@@ -0,0 +1,677 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+
+from ...modeling_utils import PreTrainedModel
+from ...utils import (
+    add_start_docstrings,
+    logging,
+)
+from .configuration_univnet import UnivNetGanConfig
+
+
+logger = logging.get_logger(__name__)
+
+
+_CHECKPOINT_FOR_DOC = "dg845/univnet-dev"
+
+UNIVNET_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "dg845/univnet-dev",
+    # See all UnivNet models at https://huggingface.co/models?filter=univnet
+]
+
+
+class UnivNetKernelPredictorResidualBlock(nn.Module):
+    """
+    Implementation of the residual block for the kernel predictor network inside each location variable convolution
+    block (LVCBlock).
+
+    Parameters:
+        channels (`int`, *optional*, defaults to 64):
+            The number of hidden channels for the residual block.
+        kernel_size (`int`, *optional*, defaults to 3):
+            The kernel size for the 1D convolutional layers in the residual block.
+        dropout (`float`, *optional*, defaults to 0.0):
+            The dropout probability for the Dropout layer in the residual block.
+        leaky_relu_slope (`float`, *optional*, defaults to 0.1):
+            The angle of the negative slope used by the leaky ReLU activation.
+    """
+
+    def __init__(
+        self,
+        channels: int = 64,
+        kernel_size: int = 3,
+        dropout: float = 0.0,
+        leaky_relu_slope: float = 0.1,
```

Comment:
I'm not clear on this suggestion, is the proposal to make something like `config: UnivNetGanConfig` an argument to `__init__` and then use e.g. `config.model_in_channels` instead of `channels`, etc.?

Reply:
yes